projects/laji/src/app/+observation/form/observation-form.component.html

```diff
@@ -101,48 +101,12 @@
 
 <!-- OWN OBSERVATIONS -->
 <section class="laji-panel" *lajiLoggedIn="true" laji-panel [title]="'observation.form.own' | translate" [autoToggle]="true" [open]="visible['own']">
-  <fieldset>
-    <div>
-      <label>
-        <laji-checkbox [(value)]="formQuery.asObserver" (valueChange)="ownItemSelected('asObserver')"></laji-checkbox>
-        {{ 'observation.form.asObserver' | translate }}
-      </label>
-      <laji-info>
-        {{ 'observation.info.asObserver' | translate }}
-      </laji-info>
-    </div>
-    <div>
-      <label>
-        <laji-checkbox [(value)]="formQuery.asEditor" (valueChange)="ownItemSelected('asEditor')"></laji-checkbox>
-        {{ 'observation.form.asEditor' | translate }}
-      </label>
-      <laji-info>
-        {{ 'observation.info.asEditor' | translate }}
-      </laji-info>
-    </div>
-  </fieldset>
-
-  <hr>
-  <div>
-    <p>{{ 'observation.intro.noOwnObservations' | translate }}</p>
-    <label><laji-checkbox [value]="formQuery.asNotEditorOrObserver" (valueChange)="ownItemSelected(ownStatutes, false)"></laji-checkbox>{{ 'observation.form.excludeOwn' | translate }}</label>
-    <laji-info>{{ 'observation.info.asNotEditororObserver' | translate }}</laji-info>
-  </div>
-
-  <ng-container *ngIf="formQuery.asEditor || formQuery.asObserver || query.qualityIssues">
-    <laji-select
-      [selected]="[query.qualityIssues]"
-      (selectedChange)="indirectQueryChange('qualityIssues', $event[0])"
-      [multiple]="false"
-      [useFilter]="false"
-      [title]="'observation.form.qualityIssues' | translate"
-      [info]="'observation.info.qualityIssues' | translate"
-      [options]="[
-        {id: 'BOTH', value: ('observation.form.qualityIssuesBOTH' | translate)},
-        {id: 'NO_ISSUES', value: ('observation.form.qualityIssuesNO_ISSUES' | translate)},
-        {id: 'ONLY_ISSUES', value: ('observation.form.qualityIssuesONLY_ISSUES' | translate)}
-      ]"></laji-select>
-  </ng-container>
+  <laji-own-observations-filter
+    [asObserverOrEditor]="getOwnObservationsFilterState()"
+    [includeQualityIssues]="getOwnQualityIssuesFilterState()"
```

Comment:
Hmm, had to remove the typing for quality issues because of this, as we can't use type casting in the template code, while a function to retrieve the type casted value would be bad for change detection. Other solutions would be a guard to do the type casting, or typing the `WarehouseQueryInterface` but that's probably problematic as it might have other possible values than what `QualityIssuesFilterState` had and maybe even as comma concatenation of that, so we'd actually need some internal model of the warehouse query interface which we'd parse into the actual query.. Both sound like an overkill.